api/v1/views/cities.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/python3
+from api.v1.views import app_views
+from flask import Flask, jsonify, abort, make_response, request
+from models import storage
+from models.state import State
+from models.city import City
+
+
+@app_views.route('/states/<state_id>/cities',
+                 methods=['GET'], strict_slashes=False)
+def get_cities_states_id(state_id):
+    """ Return the cities from a state id """
+    catch_state = storage.get('State', state_id)
+    if catch_state is None:
+        abort(404)
+    cities_list = []
+    for city in catch_state.cities:
+        cities_list.append(city.to_dict())
+    return jsonify(cities_list)
+
+
+@app_views.route('/cities/<city_id>', methods=['GET'], strict_slashes=False)
+def get_city_id(city_id):
+    """ Return the city form a id """
+    catch_city = storage.get('City', city_id)
+    if catch_city is None:
+        abort(404)
+    return jsonify(catch_city.to_dict())
+
+
+@app_views.route('/cities/<city_id>', methods=['DELETE'], strict_slashes=False)
+def del_city_id(city_id):
+    """ Delete the city form a id """
+    catch_city = storage.get('City', city_id)
+    if catch_city is None:
+        abort(404)
+    storage.delete(catch_city)
+    storage.save()
+    return jsonify({}), 200
+
+
+@app_views.route('/states/<state_id>/cities',
+                 methods=['POST'], strict_slashes=False)
+def post_cities(state_id):
```

Comment:
I saw in other scripts that this definition could use a better name. Something like "post_state_id_cities" It makes more sense regarding the route.

Reply:
Ok, I agree. Change fixed